### PR TITLE
refactor(iroh-net): keep `DerpMap` fixed

### LINKED ISSUE #1182: Change derpmap to have Arc<DerpNode> as values
The derpmap is an expensive struct yet it gets cloned around all the time.  Furthermore especially in netcheck we jump from `DerpNode` to it's name as a string to pass around so it is cheaper.  To only later look up this name and find the `DerpNode` again.  This all would be a lot nicer if we globally made the `DerpMap` contain `Arc<DerpNode>` entries.

### PR DESCRIPTION
**Pull request by dignifiedquire**
## Description

The `DerpMap` wasn't being changed anyway at runtime, only for initial config.

Closes #1182 

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.

### CODE REVIEW

File: iroh-net/src/magicsock.rs [hunk 1]
```diff
@@ -208,9 +212,8 @@ pub(self) struct Inner {
     /// If the last netcheck report, reports IPv6 to be available.
     pub(self) ipv6_reported: Arc<AtomicBool>,
 
-    // Sigh, a lock it is..
     /// None (or zero regions/nodes) means DERP is disabled.
-    pub(self) derp_map: tokio::sync::RwLock<Option<DerpMap>>,
```
Comment (rklaehn) [inline iroh-net/src/magicsock.rs:213]: 🎆

**Review by rklaehn — APPROVED** [review]
Looks good. Less locks is more better.

But why exactly don't we need set_derp_map anymore?

**Comment by dignifiedquire** [comment]
> But why exactly don't we need set_derp_map anymore?

it came from the original tailscale design, but we don't actually use it, other than on startup

**Comment by dignifiedquire** [comment]
Looks like some initial setup pieces are not quite working yet with this change..

File: iroh-net/src/derp/map.rs [hunk 2]
```diff
@@ -26,6 +27,37 @@ impl DerpMap {
         ids
     }
 
+    /// Returns an `Iteratore` over all known regions.
```
Comment (divagant-martian) [inline iroh-net/src/derp/map.rs:30]:
```suggestion
    /// Returns an `Iterator` over all known regions.
```
it felt very Italian :ok_hand:

Comment (dignifiedquire) [reply iroh-net/src/derp/map.rs:30]: 🤣

**Review by ramfox — APPROVED** [review]
I do think that in the (distant) future we will likely need something that allows us to add new derp regions to very long running nodes without restarting them, this is much simpler and all we need for now!

**Pull request merged**